pam/gdm/extension_test.go

```diff
@@ -0,0 +1,248 @@
+package gdm
+
+import (
+	"slices"
+	"testing"
+	"unsafe"
+
+	"github.com/msteinert/pam"
+	"github.com/stretchr/testify/require"
+	"github.com/ubuntu/authd/pam/pam_test"
+)
+
+func TestMain(t *testing.T) {
+	// We need to ensure that the the size of the data structures respects our
+	// expectations, so we do it at every
+	var msg jsonProtoMessage
+	require.Equal(t, int(unsafe.Sizeof(msg)), jsonProtoMessageSize,
+		"Unexpected request struct size, this is a fatal error")
+
+	require.Lessf(t, len(JSONProtoName), int(unsafe.Sizeof(msg.protocol_name)),
```

Comment:
I don't like using the `require.xxxxf` functions of the `require` package, tbh. If you [take a look at them](https://github.com/stretchr/testify/blob/master/assert/assertion_format.go), you'll see that they are just a wrapper around the normal ones and only "cast" the message as an `interface{}`. Can you switch to the "normal" ones (i.e. without the f in the end).